### PR TITLE
Local query version of Biglearn client

### PR DESCRIPTION
**Pull request by jpslav**

### CODE REVIEW

File: lib/openstax/biglearn/v1.rb [hunk 1]
```diff
@@ -85,28 +86,61 @@ def self.configuration
...
+  def self.client_name
+    Settings::Db.store.biglearn_client
   end
 
   private
 
   def self.client
+    @a_local_query_client ||= local_query_client
+    @a_real_client ||= real_client
+    @a_fake_client ||= fake_client
```
Comment (kevinburleigh75) [inline lib/openstax/biglearn/v1.rb]: Do we really want to be forcing all clients to exist/be instantiated here?

File: lib/openstax/biglearn/v1.rb [hunk 2]
```diff
@@ -85,28 +86,61 @@ def self.configuration
...
+    case client_name
+    when :local_query
+      @a_local_query_client
+    when :real
+      @a_real_client
+    when :fake
+      @a_fake_client
+    else
+      @a_real_client
+    end
```
Comment (kevinburleigh75) [inline lib/openstax/biglearn/v1.rb]: can `client_name` ever be not set? if not, else should really be an exception

File: lib/openstax/biglearn/v1.rb (hunk 1 above)
Comment (jpslav) [inline lib/openstax/biglearn/v1.rb]: good point

File: lib/openstax/biglearn/v1.rb (hunk 2 above)
Comment (jpslav) [inline lib/openstax/biglearn/v1.rb]: It should always be set unless something bad happens.  So you're probably right

**Pull request merged**